### PR TITLE
Prettifying the --time output

### PR DESCRIPTION
**Pull request by castulo**
Changing the way time is displayed when using the --time option so it
shows as a tree view so it is easier to see what items are sub-tasks
of other tasks.

Signed-off-by: Castulo Martinez <castulo.martinez@intel.com>

### CODE REVIEW

File: src/timelist.c [hunk 1]
```diff
@@ -90,8 +96,32 @@ void timelist_timer_start(timelist *head, const char *name)
...
+		number_of_spaces = (count - 1) * 4;
+	}
+	char spaces[number_of_spaces];
+	for (int i = 0; i < number_of_spaces; i++) {
+		spaces[i] = ' ';
+	}
+	spaces[number_of_spaces] = '\0';
+
+	/* build the decorated name that will be inserted */
+	string_or_die(&decorated_name, "%s%s%s", spaces, count ? "|-- " : "", name);
```
Comment (otaviobp) [inline src/timelist.c]: Just a hack using "%*s". If you do this:

    string_or_die(&decorated_name, "%*s%s%s", number_of_spaces, "", count ? "|-- " : "", name);

Comment (otaviobp) [reply src/timelist.c]: And you need to free memory allocated by string_or_die on timelist_free.

Comment (castulo) [reply src/timelist.c]: > Just a hack using "%*s". If you do this:
> 
> ```
> string_or_die(&decorated_name, "%*s%s%s", number_of_spaces, "", count ? "|-- " : "", name);
> ```

Nice, I tried doing this before but it didn't work, but know I see what I was doing wrong. 😅 Thanks for the tip

**Comment by castulo** [comment]
@otaviobp I made the changes you requested, thanks for reviewing 💪 

I would like to hold on to merging this PR until this other PR is merged though #641, I don't like submitting code without adding a test for the new code. I am not adding a test to this PR because it would duplicate the one in #641. Once #641 I will update that test with the changes in this PR and then we can merge it.

**Comment by castulo** [comment]
I rebased and updated the _bundle-add --time_ test to include this format.

**Comment by castulo** [comment]
Just removing a couple of shellcheck warnings
